docs/configuration.md

```diff
@@ -0,0 +1,143 @@
+
+## Watcher Configuration
+
+Creating a new Watcher with all defaults might look like this...
+
+```go
+watcher := gobatcher.NewWatcher(func(batch []gobatcher.IOperation) {
+    // your processing function goes here
+})
+```
+
+Creating with all available configuration options might look like this...
+
+```go
+watcher := gobatcher.NewWatcher(func(batch []gobatcher.IOperation) {
+    // your processing function goes here
+}).
+    WithMaxAttempts(3).
+    WithMaxBatchSize(500).
+    WithMaxOperationTime(1 * time.Minute)
+```
+
+- __processing_func__ [REQUIRED]: To create a new Watcher, you must provide a callback function that accepts a batch of Operations. The provided function will be called as each batch is available for processing. When the callback function is completed, it will reduce the Target by the cost of all Operations in the batch. If for some reason the processing is "stuck" in this function, they Target will be reduced after MaxOperationTime. Every time this function is called with a batch it is run as a new goroutine so anything inside could cause race conditions with the rest of your code - use atomic, sync, etc. as appropriate.
+
+- __WithMaxAttempts__ [OPTIONAL]: If there are transient errors, you can enqueue the same Operation again. If you do not provide MaxAttempts, it will allow you to enqueue as many times as you like. Instead, if you specify MaxAttempts, the Enqueue() method will return `TooManyAttemptsError{}` if you attempt to enqueue it too many times. You could check this yourself instead of just enqueuing, but this provides a simple pattern of always attempt to enqueue then handle errors.
+
+- __WithMaxBatchSize__ [OPTIONAL]: This determines the maximum number of Operations that will be raised in a single batch. This does not guarantee that batches will be of this size (constraints such rate limiting might reduce the size), but it does guarantee they will not be larger.
+
+- __WithMaxOperationTime__ [OPTIONAL]: This determines how long the system should wait for the callback function to be completed on the batch before it assumes it is done and decreases the Target anyway. It is critical that the Target reflect the current cost of outstanding Operations. The MaxOperationTime ensures that a batch isn't orphaned and continues reserving capacity long after it is no longer needed. If MaxOperationTime is not provided on the Watcher, the Batcher MaxOperationTime is used.
+
+## ProvisionedResource configuration
+
+Creating a new ProvisionedResource might look like this...
+
+```go
+resource := NewProvisionedResource(maxCapacity)
+```
+
+- __maxCapacity__ [REQUIRED]: To create a provisioned resource, you must provide the capacity. Since the ProvisionedResource is a fixed capacity rate limiter, this value serves as both MaxCapacity and Capacity.
+
+## AzureSharedResource configuration
+
+Creating a new AzureSharedResource might look like this...
+
```

Comment:
Why doesn't this section follow the same pattern as above sections by providing an "all defaults" example and an "all options" example?

Reply:
You are right! Good catch. Adjusted. Thank so much @KeithJRome